### PR TITLE
[Switch-596] Improve init and remove some modifiers

### PR DESCRIPTION
**Pull request by robergro**
Replace default values on the **init** to parameters values.
Remove some modifiers (intent, alignment and isEnabled)

### CODE REVIEW

File: core/Sources/Components/Switch/View/SwiftUI/SwitchView.swift [hunk 1]
```diff
@@ -186,33 +192,6 @@ public struct SwitchView: View {
...
-    public func intent(_ intent: SwitchIntent) -> Self {
-        self.viewModel.set(intent: intent)
-        return self
-    }
-
-    /// Set the isEnabled on switch.
-    /// - Parameters:
-    ///   - isEnabled: The state of the switch: enabled or not.
-    /// - Returns: Current Switch View.
-    public func isEnabled(_ isEnabled: Bool) -> Self {
```
Comment (michael-zimmermann) [inline core/Sources/Components/Switch/View/SwiftUI/SwitchView.swift:211]: I'd use the standard SwiftUI way of disabling components, and that is using a modifier function called `disabled`.
In UIKit it is usually a public variable called `isEnabled`.

File: core/Sources/Components/Switch/View/SwiftUI/SwitchView.swift [hunk 2]
```diff
@@ -31,19 +31,25 @@ public struct SwitchView: View {
...
+    ///   - intent: The intent of the switch.
+    ///   - alignment: The alignment of the switch.
+    ///   - isEnabled: The state of the switch: enabled or not.
     public init(
+        isOn: Binding<Bool>,
         theme: any Theme,
-        isOn: Binding<Bool>
+        intent: SwitchIntent,
+        alignment: SwitchAlignment,
+        isEnabled: Bool
```
[the remark this answers is not in the document]
Comment (robergro) [reply core/Sources/Components/Switch/View/SwiftUI/SwitchView.swift:44]: No, you're right with Michael. I update the code to have the same func as Michael does on others components (you can check the michael comment)

File: core/Sources/Components/Switch/View/SwiftUI/SwitchView.swift (hunk 1 above)
Comment (robergro) [reply core/Sources/Components/Switch/View/SwiftUI/SwitchView.swift:211]: Yeah, you're right, I update my code, remove the **isEnabled on the init** and add the **disabled func**.

File: core/Sources/Components/Switch/View/SwiftUI/SwitchView.swift [hunk 3]
```diff
@@ -31,19 +31,23 @@ public struct SwitchView: View {
...
     /// Initialize a new switch view
     /// - Parameters:
-    ///   - theme: The spark theme of the switch.
     ///   - isOn: The Binding value of the switch.
+    ///   - theme: The spark theme of the switch.
+    ///   - intent: The intent of the switch.
+    ///   - alignment: The alignment of the switch.
     public init(
+        isOn: Binding<Bool>,
         theme: any Theme,
```
Comment (michael-zimmermann) [inline core/Sources/Components/Switch/View/SwiftUI/SwitchView.swift:40]: Just wondering why you moved the isOn up?
In all other components, we have the theme as the first parameters.

**Review by michael-zimmermann — COMMENTED** [review]
Looks good to me, just one mini question

File: core/Sources/Components/Switch/View/SwiftUI/SwitchView.swift (hunk 3 above)
Comment (robergro) [reply core/Sources/Components/Switch/View/SwiftUI/SwitchView.swift:40]: Oh you're right ! 
I check the Toggle on SwiftUI and the first parameter is the isOn but I prefer respect our decision so I will update my code

**Review by michael-zimmermann — APPROVED** [review]

**Pull request merged**